src/app/zap-templates/templates/chip/helper.js

```diff
@@ -445,6 +446,130 @@ async function chip_endpoint_clusters(options)
   return templateUtil.collectBlocks(clusters, options, this);
 }
 
+/**
+ * Helper checks if the type for the bitmap is BitFlags. This generally includes
+ * all bitmaps apart from
+ * bitmap8/16/32(generally defined in types.xml)
+ * example:
+ * {{#if_is_strongly_typed_bitmap type}}
+ * strongly typed bitmap
+ * {{else}}
+ * not a strongly typed bitmap
+ * {{/if_is_strongly_typed_bitmap}}
+ *
+ * @param {*} type
+ * @returns Promise of content.
+ */
+async function if_is_strongly_typed_bitmap(type, options)
+{
+  let packageId = await templateUtil.ensureZclPackageId(this)
+  let bitmap
+  if (type && typeof type === 'string')
+  {
+    bitmap = await queryZcl.selectBitmapByName(this.global.db, packageId, type)
+  }
+  else
+  {
+    bitmap = await queryZcl.selectBitmapById(this.global.db, type)
+  }
+
+  if (bitmap) {
+    let a = await queryZcl.selectAtomicType(this.global.db, packageId, bitmap.name)
+    if (a)
+    {
+      return options.inverse(this)
```

Comment:
Again:
```suggestion
      // If this is an atomic type, it's a generic, weakly typed, bitmap.
      return options.inverse(this);
```
And I just realized: this code is written in the broken "no semicolons" style ZAP uses... please put in the semicolons; we do not want to be relying on ASI in this file.